python/pyspark/sql/dataframe.py

```diff
@@ -1238,7 +1238,7 @@ def fillna(self, value, subset=None):
             Value to replace null values with.
             If the value is a dict, then `subset` is ignored and `value` must be a mapping
             from column name (string) to replacement value. The replacement value must be
-            an int, long, float, or string.
+            an int, long, float, boolean, or string.
```

Comment:
That makes sense, I'd say that the eventual improvement would maybe be offering `fill` for bool for symetry with the rest of the types but its not necessary here rather than type checking for bool on the input.